server/mqtt.go

```diff
@@ -2415,10 +2439,12 @@ func (as *mqttAccountSessionManager) serializeRetainedMsgsForSub(rms map[string]
 		if !ok {
 			continue
 		}
+		c.mu.Lock()
```

Comment:
@kozlovic I am 0/5 on this. I noticed that everywhere else we do `c.mu.Lock` to change `sub.mqtt.prm`, so I added it here; not 100% sure if it's necessary but doesn't appear to hurt?

Reply:
The idea was that when we set that up here (for a new subscription), the callback - even if firing early - would not have access to the `prm` field because it would be blocked by the session lock (which this function runs under). But yes, I guess it is more correct to lock here.